crates/control/src/kick_target_provider.rs

```diff
@@ -0,0 +1,225 @@
+use color_eyre::Result;
+use ordered_float::NotNan;
+
+use context_attribute::context;
+use coordinate_systems::{Field, Ground};
+use framework::{AdditionalOutput, MainOutput};
+use geometry::{circle::Circle, line_segment::LineSegment, two_line_segments::TwoLineSegments};
+use linear_algebra::{distance, point, Isometry2, Point2};
+use serde::{Deserialize, Serialize};
+use types::{
+    field_dimensions::FieldDimensions, kick_target::KickTarget, obstacles::Obstacle,
+    parameters::FindKickTargetsParameters, world_state::BallState,
+};
+
+#[derive(Deserialize, Serialize)]
+pub struct KickTargetProvider {}
+
+#[context]
+pub struct CreationContext {}
+
+#[context]
+pub struct CycleContext {
+    ball_state: RequiredInput<Option<BallState>, "ball_state?">,
+    ground_to_field: RequiredInput<Option<Isometry2<Ground, Field>>, "ground_to_field?">,
+    obstacles: Input<Vec<Obstacle>, "obstacles">,
+
+    field_dimensions: Parameter<FieldDimensions, "field_dimensions">,
+
+    ball_radius_for_kick_target_selection:
+        Parameter<f32, "kick_target_provider.ball_radius_for_kick_target_selection">,
+    find_kick_targets:
+        Parameter<FindKickTargetsParameters, "kick_target_provider.find_kick_targets">,
+    max_kick_around_obstacle_angle:
+        Parameter<f32, "kick_target_provider.max_kick_around_obstacle_angle">,
+
+    corner_kick_strength: Parameter<f32, "kick_target_provider.corner_kick_strength">,
+    kick_targets: AdditionalOutput<Vec<KickTarget>, "kick_targets">,
```

Comment:
Since this is now always required by the kick selector, is there a reason to have this still be an `AdditionalOutput`?

Reply:
Done

Reply:
No there is not. Fixed